neutrons/reflectivity_ui
# Dead-time correction algorithm and UI elements for settings

# Short description of the changes:
This branch combines back-end changes by @mdoucet in the branch: https://github.com/neutrons/reflectivity_ui/tree/dead_time
and adds UI changes by @searscr and @backmari.

# Long description of the changes:
- new Mantid-style algorithm `SingleReadoutDeadTimeCorrection`
- logic to apply dead-time correction when a file is loaded
- UI elements to configure dead-time correction settings (applied globally for all runs)
- logic to reload files upon changes in the dead-time correction settings

**TODO:** add unit testing of the new algorithm `SingleReadoutDeadTimeCorrection`

# Check list for the pull request

# Check list for the reviewer

# Manual test for the reviewer
1. Open the file 
[REF_M_42112+42113_Specular_SA_test_dead-time_datarepo.dat.txt](https://github.com/user-attachments/files/16513446/REF_M_42112%2B42113_Specular_SA_test_dead-time_datarepo.dat.txt)
and edit the 4 file paths to point to the location of your data repo folder. (Can remove the .txt filename suffix added to fool GitHub.)
2. Load the reduced file in QuickNXS.
3. Verify that the "Dead Time Correction" options are available in the UI and test that checking/unchecking the checkbox enables/disables the "Settings" button.
4. The files in the reduction table and direct beam table need to be reloaded when the dead-time settings are changed, and the user is asked to confirm before reloading. Verify that all other configuration stays the same when dead-time settings are changed.
5. Verify that the reflectivity plot is updated when dead-time settings are changed.

![image](https://github.com/user-attachments/assets/bf324af0-e1d7-4e9b-9ffc-ddf72c951f3b)

# References
[Story 5023: [QUICKNXS] Add dead time options to UI](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=5023)
